backend/rates/history.go

```diff
@@ -178,7 +178,8 @@ func (updater *RateUpdater) updateHistory(ctx context.Context, coin, fiat string
 
 	updater.historyMu.Lock()
 	defer updater.historyMu.Unlock()
-	allRates := append(updater.history[bucketName], fetchedRates...)
+	updater.history[bucketName] = append(updater.history[bucketName], fetchedRates...)
+	allRates := updater.history[bucketName]
 	sort.Slice(allRates, func(i, j int) bool {
 		return allRates[i].timestamp.Before(allRates[j].timestamp)
 	})
```

Comment:
the line below this one is not needed I think. For clarity maybe it's better to just do this?

```
	updater.history[bucketName] = append(updater.history[bucketName], fetchedRates...)
	sort.Slice(updater.history[bucketName], func(i, j int) bool {
		return updater.history[bucketName][i].timestamp.Before(updater.history[bucketName][j].timestamp)
	})
	return len(fetchedRates), nil
```

wdyt?

Reply:
Yes, this is a better solution imho. Also the reassignment `updater.history[bucketName] = allRates` is not useful and just more memory association. I've applied your change.